src/devices/Media/SoundDevice/Devices/UnixSoundDevice.cs

```diff
@@ -0,0 +1,551 @@
+                ByteRate = Settings.RecordingSampleRate * Settings.RecordingBitsPerSample * Settings.RecordingChannels / 8,
+                BlockAlign = (ushort)(Settings.RecordingBitsPerSample * Settings.RecordingChannels / 8),
+                BitsPerSample = Settings.RecordingBitsPerSample,
+                Subchunk2Id = new[] { 'd', 'a', 't', 'a' },
+                Subchunk2Size = second * Settings.RecordingSampleRate * Settings.RecordingBitsPerSample * Settings.RecordingChannels / 8
+            };
+
+            SetWavHeader(saveStream, header);
+
+            OpenRecordingPcm();
+            PcmInitialize(_recordingPcm, header, ref @params, ref dir);
+            ReadStream(saveStream, header, ref @params, ref dir);
+            CloseRecordingPcm();
+        }
+
+        private void SetWavHeader(Stream wavStream, WavHeader header)
+        {
+            Span<byte> writeBuffer2 = stackalloc byte[2];
+            Span<byte> writeBuffer4 = stackalloc byte[4];
+
+            wavStream.Position = 0;
+
+            try
+            {
+                Encoding.ASCII.GetBytes(header.ChunkId, writeBuffer4);
+                wavStream.Write(writeBuffer4);
+
+                BinaryPrimitives.WriteUInt32LittleEndian(writeBuffer4, header.ChunkSize);
+                wavStream.Write(writeBuffer4);
+
+                Encoding.ASCII.GetBytes(header.Format, writeBuffer4);
+                wavStream.Write(writeBuffer4);
+
+                Encoding.ASCII.GetBytes(header.Subchunk1ID, writeBuffer4);
+                wavStream.Write(writeBuffer4);
+
+                BinaryPrimitives.WriteUInt32LittleEndian(writeBuffer4, header.Subchunk1Size);
+                wavStream.Write(writeBuffer4);
+
+                BinaryPrimitives.WriteUInt16LittleEndian(writeBuffer2, header.AudioFormat);
+                wavStream.Write(writeBuffer2);
+
+                BinaryPrimitives.WriteUInt16LittleEndian(writeBuffer2, header.NumChannels);
+                wavStream.Write(writeBuffer2);
+
+                BinaryPrimitives.WriteUInt32LittleEndian(writeBuffer4, header.SampleRate);
+                wavStream.Write(writeBuffer4);
+
+                BinaryPrimitives.WriteUInt32LittleEndian(writeBuffer4, header.ByteRate);
+                wavStream.Write(writeBuffer4);
+
+                BinaryPrimitives.WriteUInt16LittleEndian(writeBuffer2, header.BlockAlign);
+                wavStream.Write(writeBuffer2);
+
+                BinaryPrimitives.WriteUInt16LittleEndian(writeBuffer2, header.BitsPerSample);
+                wavStream.Write(writeBuffer2);
+
+                Encoding.ASCII.GetBytes(header.Subchunk2Id, writeBuffer4);
+                wavStream.Write(writeBuffer4);
+
+                BinaryPrimitives.WriteUInt32LittleEndian(writeBuffer4, header.Subchunk2Size);
+                wavStream.Write(writeBuffer4);
+            }
+            catch
+            {
+                throw new Exception("Write WAV header error.");
```

Comment:
we should either let the exception bubble up or wrap it (set inner exception to original) with some better typed exception

Reply:
I will remove try-catch